spec/models/user_spec.rb

```diff
@@ -391,6 +405,27 @@
     end
   end
 
+  describe '.reset_availability!' do
+    let!(:user) { create(:user, :available) }
+    let!(:user_interested_in_coaching) { create(:user, :available, :interested_in_coaching) }
+
+    before do
+      User.reset_availability!
```

Comment:
I feel like doing this in a before block reduces the possibilities to test the actual behavior of the method.

I would suggest to rather test the "change" and its impact:
```ruby
expect { User.reset_availablity! }.to change { user.reload.availablity }.from(true).to(false)
```